swap/swap.go

```diff
@@ -646,31 +667,28 @@ func (s *Swap) bindToContractAt(address common.Address) (contract.Contract, erro
 }
 
 // Deploy deploys the Swap contract
-func (s *Swap) Deploy(ctx context.Context, initialDepositAmount uint64) (contract.Contract, error) {
+func (s *Swap) Deploy(ctx context.Context) (contract.Contract, error) {
 	opts := bind.NewKeyedTransactor(s.owner.privateKey)
-	// initial topup value
-	opts.Value = big.NewInt(int64(initialDepositAmount))
 	opts.Context = ctx
-	swapLog.Info("Deploying new swap", "owner", opts.From.Hex(), "deposit", opts.Value)
-	return s.deployLoop(opts, defaultHarddepositTimeoutDuration)
+	swapLog.Info("Deploying new swap", "owner", opts.From.Hex())
+	chequebook, err := s.chequebookFactory.DeploySimpleSwap(opts, s.owner.address, big.NewInt(int64(defaultHarddepositTimeoutDuration)))
```

Comment:
Why is the loop not needed anymore?

Reply:
I think it was never needed. If it fails it fails and there is not really a possibility that it would magically not fail anymore after 5 seconds waiting.

However, @ralph-pichler recently did some tests on RSK and it failed the first time but not on subsequent tries?

Reply:
Yes, but I believe that's an issue on the rsk side. Usually a retry is probably worse than aborting (e.g. trying again, still failing and wasting gas).

Reply:
The loop is actually a remnant of the legacy code...maybe it was waiting for confirmation or something...

In any case if you say that it is not needed I am cool.